File: examples/autoreply.py
"""Minimal auto-reply agent.

Point CASPIAN_BASE_URL and CASPIAN_API_KEY (or the legacy COMM_BASE_URL /
COMM_API_KEY names) at a Caspian gateway (hosted, or any deployment of the
Caspian gateway) — `caspian init` writes them to ./.env — then:

    uv run python examples/autoreply.py
"""

from caspian_sdk import CommClient

# Reads CASPIAN_API_KEY / CASPIAN_BASE_URL (legacy COMM_API_KEY / COMM_BASE_URL
# also work) from the environment or ./.env, e.g. as written by `caspian init`
# (base_url defaults to the hosted gateway at https://api.trycaspianai.com).
client = CommClient()

customer = client.create_customer("Acme")
agent = client.create_agent("Support Agent")
connection = client.connect_email(customer["id"], agent["id"], username="support")
print(f"Email connection active: {connection['address']}")


@client.on_message
def handle(message):
    print(f"Inbound from {message.sender['address']}: {message.text!r}")
    message.reply(f"Thanks for reaching out. You said: {message.text}")


print("Listening for inbound messages (Ctrl+C to stop)")
client.listen()

File: examples/autoreply.ts
/**
 * Minimal auto-reply agent (TypeScript).
 *
 * TS equivalent of autoreply.py in this folder: connect an email inbox,
 * register a message handler, reply, and listen.
 *
 * Setup (no build/publish step required — runs straight off SDK source):
 *   export COMM_API_KEY=your_key_here     # or CASPIAN_API_KEY
 *   npx tsx examples/autoreply.ts
 *
 * COMM_BASE_URL (or CASPIAN_BASE_URL) is optional; it defaults to the
 * hosted gateway at https://api.trycaspianai.com.
 */
import { CommClient } from "../sdks/typescript/src/index.js";

// Reads COMM_API_KEY / COMM_BASE_URL (or the CASPIAN_* equivalents) from the
// environment or a local .env file - no arguments required.
const client = new CommClient();

async function main() {
  const inbox = await client.connectEmail();
  console.log(`Email connection active: ${inbox.address}`);

  client.onMessage(async (message) => {
    const from = String(message.sender?.["address"] ?? "unknown sender");
    console.log(`Inbound from ${from}: ${message.text}`);
    await message.reply(`Thanks for reaching out. You said: ${message.text}`);
  });

  console.log("Listening for inbound messages (Ctrl+C to stop)");
  await client.listen();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});